encfs/main.cpp

```diff
@@ -575,6 +607,18 @@ int main(int argc, char *argv[]) {
     return EXIT_FAILURE;
   }
 
+  // Let's unmount if requested
+  if (encfsArgs->opts->unmount) {
+    fuse_unmount(encfsArgs->opts->mountPoint.c_str(), nullptr);
```

Comment:
Just tested this on Linux: If the filesystem is busy, the filesystem is lazy-umounted. This means a shell running inside the mountpoint can still access all files, and the encfs process keeps running.

This is unexpected and we should return an error instead

Reply:
Is the fuse_unmount function taking two parameter deprecated? Does not show up here https://libfuse.github.io/doxygen/fuse_8h.html#aade4d93e323c341fe2859d759e9cb8d8

Reply:
> Just tested this on Linux: If the filesystem is busy, the filesystem is lazy-umounted.

Well this is a Fuse behavior actually, this is how `fuse_unmount` works...

> This is unexpected and we should return an error instead

We can't as explained above, as `fuse_unmount` returns void and succeeds.

> Is the fuse_unmount function taking two parameter deprecated? Does not show up here

Yep your like is about Fuse 3.
`fuse_unmount` with 2 parameters is a Fuse 2 call.
You can see the changes here :
https://github.com/libfuse/libfuse/releases/tag/fuse-3.0.0
(search for fuse_unmount)

Reply:
I personally wouldn't  mind dropping `fuse_unmount()` from the code, mainly because I've no use for it. ;-)

Reply:
We need it in any case for the automatic unmount on idle